winrm/x509.go

```diff
@@ -0,0 +1,280 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package winrm
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/juju/utils"
+	"github.com/juju/utils/cert"
+)
+
+// List of exported internal errors
+var (
+	ErrNoClientCert       = errors.New("No client winrm cert inside dir")
+	ErrNoClientPrivateKey = errors.New("No client private key inside dir")
+	ErrNoX509Folder       = errors.New("No configuration x509 folder is present in this location")
+)
+
+// x509WinRM type that defines windows remote manager
+// credentials for client-server secure communication
+type x509 struct {
+	cert   []byte // client cert
+	key    []byte // client private key
+	cacert []byte // ca server cert
+	err    error  // internal err state
+}
+
+var (
+	mu          sync.Mutex
+	credentials x509 //winrm credentials
+)
+
+// LoadClientCert generates client cert for x509 authentication
+// if the directory in the coresponding paths key, cert
+// if the base dir dosen't exist it will create and generate the key/cert pair
+// LoadClientCert function is lock protected allowing multiple threads to call this function
+func LoadClientCert(key, cert string) error {
```

Comment:
Please call them "keyFile", "certFile" or "keyPath", "certPath". Named as they are, I would expect them contain the key/cert values.